src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -363,4 +416,79 @@ protected int doHashCode() {
     public String getWriteableName() {
         return NAME;
     }
+
+    /**
+     * Builder for KNNQueryBuilder class to create KNNQueryBuilder, which can be used to build k-NN query or r-NN query.
+     * */
+    public static class Builder {
```

Comment:
For builder pattern, why do we need to create a new class instead of adding builder type methods on KNNQueryBuilder? Here is an example of how it looks for BoolQueryBuilder: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/query/BoolQueryBuilder.java#L136-L146

Reply:
Adopting the class of builder pattern, I think it's better in code readability and adding parameter validation. In the future, we can easily add score type parameter here as well.

Reply:
What do you mean? I am saying we dont need a Builder class because KNNQueryBuilder is already a builder

Reply:
Agree. We don't need new class. We can use move all these methods to KNNQueryBuilder

Reply:
Moved them to KNNQueryBuilder